File: beautify_api/8.0.0/aspnet-core/src/BanHangBeautify.Application/NhanSu/NhanVien/Dto/CreateOrEditNhanSuDto.cs
﻿using Abp.Application.Services.Dto;
using BanHangBeautify.Consts;
using BanHangBeautify.NewFolder;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace BanHangBeautify.NhanSu.NhanVien.Dto
{
    public class CreateOrEditNhanSuDto : EntityDto<Guid>
    {
        [MaxLength(50)]
        public string MaNhanVien { get; set; }
        [MaxLength(20)]
        public string Ho { set; get; }
        [MaxLength(50)]
        public string TenLot { set; get; }
        [MaxLength(256)]
        [Required(ErrorMessage = "Tên nhân viên không được để trống")]
        public string TenNhanVien { get; set; }
        [MaxLength(2000)]
        public string DiaChi { get; set; }
        [MaxLength(256)]
        public string SoDienThoai { get; set; }
        [MaxLength(256)]
        public string CCCD { get; set; }
        public DateTime? NgaySinh { get; set; }
        public byte? KieuNgaySinh { get; set; } = 0;
        public byte? GioiTinh { get; set; }
        [MaxLength(256)]
        public string NgayCap { get; set; }
        [MaxLength(2000)]
        public string NoiCap { get; set; }
        public string Avatar { get; set; }
        public AvatarFile AvatarFile { set; get; }

        //public Guid IdPhongBan { get; set; }
        public Guid? IdChiNhanh { set; get; }
        public Guid? IdChucVu { set; get; }
        public string GhiChu { get; set; }
        public int TrangThai { get; set; } = TrangThaiNhanVienConst.Ranh;
        public List<Guid>? Services { get; set; }

    }
}

File: beautify_app/lib/screens/app/nhan_vien/models/CreateOrEditNhanSuDto.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class CreateOrEditNhanSuDto {
  late String? id;
  late String? maNhanVien;
  late String? tenNhanVien;
  late String? diaChi;
  late String? soDienThoai;
  late String? cccd;
  late String? ngaySinh;
  late int? kieuNgaySinh;
  late int? gioiTinh;
  late String? ngayCap;
  late String? noiCap;
  late String? avatar;
  late String? idChucVu;
  late String? ghiChu;
  CreateOrEditNhanSuDto({
    this.id = '3fa85f64-5717-4562-b3fc-2c963f66afa6',
    this.maNhanVien = "NS000",
    this.tenNhanVien,
    this.diaChi,
    this.soDienThoai,
    this.cccd,
    this.ngaySinh,
    this.kieuNgaySinh = 0,
    this.gioiTinh = 0,
    this.ngayCap,
    this.noiCap,
    this.avatar,
    this.idChucVu,
    this.ghiChu,
  });

  CreateOrEditNhanSuDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    maNhanVien = json['maNhanVien'];
    tenNhanVien = json['tenNhanVien'];
    diaChi = json['diaChi'];
    soDienThoai = json['soDienThoai'];
    cccd = json['cccd'];
    ngaySinh = json['ngaySinh'];
    kieuNgaySinh = json['kieuNgaySinh'];
    gioiTinh = json['gioiTinh'];
    ngayCap = json['ngayCap'];
    noiCap = json['noiCap'];
    avatar = json['avatar'];
    idChucVu = json['idChucVu'];
    ghiChu = json['ghiChu'];
  }

  String toJson() => json.encode(toMap());

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'id': id,
      'maNhanVien': maNhanVien,
      'tenNhanVien': tenNhanVien,
      'diaChi': diaChi,
      'soDienThoai': soDienThoai,
      'cccd': cccd,
      'ngaySinh': ngaySinh,
      'kieuNgaySinh': kieuNgaySinh,
      'gioiTinh': gioiTinh,
      'ngayCap': ngayCap,
      'noiCap': noiCap,
      'avatar': avatar,
      'idChucVu': idChucVu,
      'ghiChu': ghiChu
    };
  }
}
